src/library/module_mgr.cpp

```diff
@@ -130,11 +130,7 @@ static module_id resolve(search_path const & path,
                          module_id const & module_file_name,
                          module_name const & ref) {
     auto base_dir = dirname(module_file_name);
-    try {
-        return find_file(path, base_dir, ref.m_relative, ref.m_name, ".lean");
-    } catch (lean_file_not_found_exception) {
-        return olean_file_to_lean_file(find_file(path, base_dir, ref.m_relative, ref.m_name, ".olean"));
-    }
+    return find_file(path, base_dir, ref.m_relative, ref.m_name, ".lean");
 }
```

Comment:
I'm concerned we may be creating a problem. I believe it is reasonable to expect that some users, in a distant future, will want to distribute `.olean`  files without the corresponding `.lean` files.

Reply:
This change, together with the version check change, had the goal of making .olean files a transparent optimization (and fixing #1828). But I don't have a strong opinion on it, I can remove it from this PR.

Reply:
> I don't have a strong opinion on it, I can remove it from this PR.

I don't have a strong opinion either, but I'm still concerned. I feel we will regret this decision in the future. Regarding #1828, I think this is a nitpicking issue. We could add a `--clean` option to `leanpkg`, and close the issue.

Reply:
Here's another possible concern: Do we (ever) want to separate Lean source and object files into separate directories? For e.g. C++ this seems to be a good approach, but does it make sense when object files can also act as source files?

Reply:
For example, my home directory at the university is on a network filesystem. I don't store Lean in it because I'm concerned about the size and frequent writes to build files. For the C++ build directory, moving it to the local drive is no problem at all.

Reply:
I've removed the commit for now